download_py_package.py

```diff
@@ -35,6 +38,39 @@
 
 def download_py_package():
     """Download package which needs to be analyzed by future steps."""
+    is_analyzable = IsPackageSIAnalyzableMessage()
+    missing_version = MissingVersionMessage()
+
+    page = request.urlopen(os.path.join(Configuration.PACKAGE_INDEX, Configuration.PACKAGE_NAME))
+    version_exists = False
+
+    for link in BeautifulSoup(
+        page, "html.parser", from_encoding=page.info().get_param("charset"), parse_only=SoupStrainer("a")
+    ):
+        if link.string.endswith(f"-{Configuration.PACKAGE_VERSION}.zip") or link.string.endswith(
+            f"-{Configuration.PACKAGE_VERSION}.tar.gz"
+        ):
+            break
+        elif f"-{Configuration.PACKAGE_VERSION}-" in link.string:
+            version_exists = True
+    else:
+        if version_exists:
+            message_contents = is_analyzable.MessageContents(
+                package_name=Configuration.PACKAGE_NAME,
+                package_version=Configuration.PACKAGE_VERSION,
+                index_url=Configuration.PACKAGE_INDEX,
+                value=False,
+            )
+            is_analyzable.publish_to_topic(message_contents)
```

Comment:
don't we store message or messages so that that we can reuse send-message/send-messages tasks with thoth-messaging CLI?

Reply:
Yeah, that makes sense.  We'll probably have to use something like this https://github.com/argoproj/argo/blob/master/examples/dag-continue-on-fail.yaml in `thoth-application`.  In the workflow we can then check the status of a specific step.

```python
if {{workflow.tasks.<taskA>.status}} == failure:
  send_message(task)
else:
  continue_dag_execution()  # check task status for immediate next tasks here too, continue_on_failure treats the task as completed
```

Reply:
^^^^^
this will have to be converted to argo-workflow yaml but this looks a lot simpler rn

Reply:
can we just put on top of si-bandit and si cloc task `when {{workflow.tasks.download-package.status}} == Succeeded` and after the download package this new task together with the one that send message with `when {{workflow.tasks.download-package.status}} == Failed`